slash/log.py

```diff
@@ -212,13 +207,30 @@ def _error_added_filter(record, handler): # pylint: disable=unused-argument
         handler, log_path = self._get_file_log_handler(path, symlink=None, bubble=True, filter=_error_added_filter)
         if log_path and self.session.results.current is self.session.results.global_result:
             self.session.results.global_result.add_extra_log_path(log_path)
-        return handler.applicationbound()
+        return handler, log_path
 
     def _get_silenced_logs_context(self):
         if not config.root.log.silence_loggers:
             return ExitStack()
         return SilencedLoggersHandler(config.root.log.silence_loggers).applicationbound()
 
+    @contextmanager
+    def _wrap_with_cleanup(self, ctx, path):
```

Comment:
I think it should be called `_deleting_files_if_needed(handler)` and deduce the path inside